lib/rules/no-unstable-nested-components.js

```diff
@@ -0,0 +1,434 @@
+/**
+ * @fileoverview Prevent creating unstable components inside components
+ * @author Ari Perkkiö
+ */
+
+'use strict';
+
+const Components = require('../util/Components');
+const docsUrl = require('../util/docsUrl');
+
+// ------------------------------------------------------------------------------
+// Constants
+// ------------------------------------------------------------------------------
+
+const ERROR_MESSAGE_WITHOUT_NAME = 'Declare this component outside parent component or memoize it.';
+const COMPONENT_AS_PROPS_INFO = ' If you want to allow component creation in props, set allowAsProps option to true.';
+const RENDER_REGEXP = /^render/;
```

Comment:
instead of a regex, can we use `.startsWith` here? https://npmjs.com/string.prototype.startswith if it's not in supported node versions.

Reply:
`startsWith` can be used without polyfill. [According to MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith) it has been introduced in v4 which is the lowest version supported by `eslint-plugin-react`. 

I also verified this in container just to be sure:
```bash
> process.version
'v4.0.0'
> 'temp'.startsWith('te')
true
```